libs/scripts/system.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/sh
+#set -e
+GO_VERSION=$1
+ROCKSDB_VERSION=$2
+
+get_distribution() {
+  lsb_dist=""
+  # Every system that we officially support has /etc/os-release
+  if [ -r /etc/os-release ]; then
+    lsb_dist="$(. /etc/os-release && echo "$ID")"
+  fi
+  # Returning an empty string here should be alright since the
+  # case statements don't act unless you provide an actual value
+  echo "$lsb_dist"
+}
+
+is_darwin() {
+  case "$(uname -s)" in
+  *darwin*) true ;;
+  *Darwin*) true ;;
+  *) false ;;
+  esac
+}
+
+get_system_version() {
+  lsb_dist=$(get_distribution)
+  lsb_dist="$(echo "$lsb_dist" | tr '[:upper:]' '[:lower:]')"
+  system_version=
+  case "$lsb_dist" in
+  ubuntu)
+    system_version="ubuntu"
+    ;;
+  centos)
+    system_version="centos"
+    ;;
+  alpine)
+    system_version="alpine"
+    ;;
+  *)
+    if [ -z "$lsb_dist" ]; then
+      if is_darwin; then
+        system_version="macos"
+      fi
+    else
+      echo
+      echo "ERROR: Unsupported system '$lsb_dist', only support centos,ubuntu,alpine,macos"
```

Comment:
Windows os and some cloud os can not be supported? This will result in okbchain only compiling and running on centos,ubuntu,alpine,macos.

Reply:
windows we have not support last year by check_version(Makfile).  So we may not support windows